config/opal_setup_cc.m4

```diff
@@ -23,6 +23,122 @@ dnl
+
+             AS_IF([test $opal_prog_cc_c11_helper_atomic_var_available = yes], [opal_proc_cc_c11_helper_tmp=1],
+                   [opal_proc_cc_c11_helper_tmp=0])
+
+             AC_DEFINE_UNQUOTED([OPAL_C_HAVE_C11_ATOMIC], [$opal_proc_cc_c11_helper_tmp], [Whether C compiler supports C11 atomics])
+
+             AS_IF([test $opal_prog_cc_c11_helper__atomic_available = yes], [opal_proc_cc_c11_helper_tmp=1],
+                   [opal_proc_cc_c11_helper_tmp=0])
+
+             AC_DEFINE_UNQUOTED([OPAL_C_HAVE__ATOMIC], [$opal_proc_cc_c11_helper_tmp], [Whether C compiler supports __Atomic])
+
+             dnl At this time Open MPI only requires thread local and the atomic convenience types
+             AS_IF([test $opal_prog_cc_c11_helper__thread_local_available = yes && test $opal_prog_cc_c11_helper_atomic_var_available = yes],
+                   [$2], [$3])
+
+             CFLAGS=$opal_prog_cc_c11_helper_CFLAGS_save
+
+             OPAL_VAR_SCOPE_POP
+])
+
+AC_DEFUN([OPAL_PROG_CC_C11],[
+             if test -z $opal_cv_c11_supported ; then
+                 opal_cv_c11_supported=no
+
+                 AC_MSG_NOTICE([checking if $CC supports C11 without a flag])
+                 OPAL_PROG_CC_C11_HELPER([], [opal_cv_c11_noflag=yes], [opal_cv_c11_noflag=no])
+
+                 # some compilers pass the C11 no-flag test but fail to compile C99
+                 if test "x$opal_cv_c11_noflag" = "xyes" ; then
+                    AC_MSG_CHECKING([if C99 works without a flag])
+
+                    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+#if __STDC_VERSION__ < 199409L
+#error "Need a flag for C99"
+#endif
+                                                       ]],[])],
+                                      [opal_proc_cc_c99_flag_need=yes],[opal_proc_cc_c99_flag_need=no])
+
+                    AC_MSG_RESULT([$opal_proc_cc_c99_flag_need])
+
+                    if test $opal_proc_cc_c99_flag_need = no ; then
+                        # need a flag for c11
+                        opal_cv_c11_noflag=no
+                    fi
+                 fi
+
+                 if test "x$opal_cv_c11_noflag" = "xno" ; then
+                     flags="-std=gnu11 -std=c11 -c11"
+
+
+                     AC_MSG_NOTICE([checking if $CC supports C11 with a flag])
+                     opal_cv_c11_flag=
+                     for flag in $(echo $flags | tr ' ' '\n') ; do
+                         OPAL_PROG_CC_C11_HELPER([$flag],[opal_cv_c11_flag=$flag],[])
+                         if test ! "x$opal_cv_c11_flag" = x ; then
+                             CFLAGS="$CFLAGS $opal_cv_c11_flag"
+                             AC_MSG_NOTICE([using $flag to enable C11 support])
+                             opal_cv_c11_supported=yes
+                             break;
+                         fi
+                     done
+
+                     if ! test -z "$opal_cv_c11_flag" ; then
```

Comment:
You used the `x` test notation above, but used `-z` here.  Please be consistent.